guest/file_manager.go

```diff
@@ -167,6 +167,11 @@ func (m FileManager) TransferURL(ctx context.Context, u string) (*url.URL, error
 		return nil, err
 	}
 
+	if host.Config == nil {
+		// SDK doc says "This might not be available for a disconnected host."
+		return turl, nil
```

Comment:
Shouldn't this return an error? How useful is `turl` in this case?

Reply:
If the host is actually running in this case, turl can still work.  host.Config is used for 2 things:

1) The hostname in `turl` is valid in most cases, but we try here to "prefer an ESX management IP" as there have been a few cases where the hostname cannot be resolved (DNS) depending on where the client code is running.

2) soap.Client.SetThumbprint - the client may already trust the host's cert, have already saved its thumbprint via a previous call or have `InsecureSkipVerify=true`

Reply:
Thank you for explaining this.

Reply:
Added the comments above and more to the code itself, do you mind reviewing the comments?